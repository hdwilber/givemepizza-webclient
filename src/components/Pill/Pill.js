import React from 'react'
import PropTypes from 'prop-types'
import _noop from 'lodash/noop'
import styles from './styles.module.css'

class Pill extends React.PureComponent {
  render() {
    const { text, onRemove } = this.props
    return (
      <div className={styles.pill}>
        <span>{ text }</span>
        <span 
          title="remove"
          className={styles.remove}
          onClick={e => {
            e.preventDefault()
            e.stopPropagation()
            onRemove()
          }}
        >
          x
        </span>
      </div>
    )
  }
}
Pill.propTypes = { 
  text: PropTypes.string,
  onRemove: PropTypes.func,
}

Pill.defaultProps = {
  onRemove: _noop,
}

export default Pill
