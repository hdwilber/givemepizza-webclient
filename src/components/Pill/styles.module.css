.pill {
	background-color: gray;
	color: lightgray;
	border-radius: 5px;
}

.remove {
	font-size: 0.5rem;
	display: inline-block;
}
.remove:hover {
	background-color: red;
}
