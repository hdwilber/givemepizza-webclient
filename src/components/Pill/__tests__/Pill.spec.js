import React from 'react'
import { mount } from 'enzyme'
import Pill from '../Pill'

function setupProps() {
  return {
    text: 'Pill example',
    onRemove: jest.fn(),
  }
}

function setup(props) {
  return mount(<Pill {...props} />)
}

describe('<Pill />', () => {
  let props
  let wrapper
  beforeEach(() => {
    props = setupProps()
    wrapper = setup(props)
  })
  it('should match snapshot', () => {
    const instance = <Pill {...props} />
    expect(instance).toMatchSnapshot()
  })
  it('should call onRemove when the X label is clicked', () => {
    const xWrapper = wrapper.find('span').at(1)
    xWrapper.simulate('click', { 
      preventDefault: jest.fn(),
      stopPropagation: jest.fn(),
    })
    expect(props.onRemove).toHaveBeenCalledTimes(1)
  })
  it('should render correctly the text', () => {
    const textWrapper = wrapper.find('span').at(0)
    expect(textWrapper.text()).toEqual(props.text)
  })
})
