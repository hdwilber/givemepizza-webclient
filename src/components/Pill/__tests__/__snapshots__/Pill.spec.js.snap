// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pill /> should match snapshot 1`] = `
<Pill
  onRemove={[MockFunction]}
  text="Pill example"
/>
`;
